lib/matplotlib/axis.py

```diff
@@ -1326,7 +1328,8 @@ def get_minorticklocs(self):
         tol = (hi - lo) * 1e-5
```

Comment:
While we are here,  I think the tolerance should be smaller here. .... 1.0001 == 1.000001 with this....  I would tend to go w/ something like `1e-10`.

Reply:
This is in pixel space though.  If the goal is to prevent tick/grid/label collisions shouldn't we make this closer to 1?

Reply:
I don't think this is a visual issue, but rather a math one.  If the user wants a major tick at 1.0 and a minor tick at 1.00001 I don't think we should stop them (or make them use a kwarg) short of numerical roundoff.

Reply:
All of the complaints about this have been due to visual overlap.

The problem we have is that we can not tell "the user intentionally put a major and minor ticks on top of each other so we should do what we are told" (ex, @ImportanceOfBeingErnest date example) vs "the user accidentally set up locators that collide and we should do the obviously correct thing and de-conflict them"

Previously the behavior was the first (which when we made the log minor locator smarter caused problems and @ImportanceOfBeingErnest was relying on) but now we do the second.  If we accept that we want to de-conflict the ticks as a primarily a display issue then we should do the filtering based on how close it comes in pixel space.

Using this flag the user can pick between the two behavior (by telling us which of the two cases they want) and we chose to default to door number 2  ("do 'the right thing' visually) as I think that is going to be the more common intention.

Reply:
Thats fine, I understand your point.  However: 

> then we should do the filtering based on how close it comes in pixel space.

I don't think we should do any deciding of what to send to the renderer in pixel space because that means a 100-dpi figure is different than 200 dpi figure.  You will end up with the inconsistent situation that a pdf that has 72 dpi virtual pixels will have fewer ticks than a png with 100 dpi.  

If your argument were instead to do the deconflict as a fraction of tick width, I'd be fine w/ that.  linewidths are in physical units.

Anyway, this comment wasn't meant at all to be a blocker on this PR.  The overall idea seems fine to me.

Reply:
The DPI dependence is a very good point.  I would be comfortable tweaking the exact de-confliction criteria later.